Sources/Sentry/SentryProfiler.mm

```diff
@@ -358,21 +341,21 @@ + (BOOL)isCurrentlyProfiling
 + (nullable SentryEnvelopeItem *)createProfilingEnvelopeItemForTransaction:
     (SentryTransaction *)transaction
 {
-    const auto profiler = profilerForFinishedTracer(transaction.trace);
-    if (!profiler) {
-        SENTRY_LOG_WARN(@"Expected a profiler for tracer id %@ but none was found",
-            transaction.trace.traceId.sentryIdString);
+    const auto payload = [self collectProfileBetween:transaction.startSystemTime
+                                                 and:transaction.endSystemTime
+                                            forTrace:transaction.trace.traceId
+                                               onHub:transaction.trace.hub];
```

Comment:
Similar thought here w.r.t. parameters, why not just pass the transaction? This is more repetitive and verbose.